tests/unit/test_bigquery_adapter.py

```diff
@@ -340,12 +345,12 @@ def test_acquire_connection_maximum_bytes_billed(self, mock_open_connection):
 
     def test_cancel_open_connections_empty(self):
         adapter = self.get_adapter("oauth")
-        self.assertEqual(adapter.cancel_open_connections(), None)
+        self.assertEqual(len(list(adapter.cancel_open_connections())), 0)
 
     def test_cancel_open_connections_master(self):
         adapter = self.get_adapter("oauth")
-        adapter.connections.thread_connections[0] = object()
-        self.assertEqual(adapter.cancel_open_connections(), None)
+        adapter.connections.thread_connections[0] = mock_connection("master")
+        self.assertEqual(len(list(adapter.cancel_open_connections())), 0)
```

Comment:
this is currently failing as the len is showing 1 meaning a job isn't being dropped because its an empty list because this is testing dropping connections not jobs is there a loophole we need to add to our above logic in `cancel_open`

Reply:
I think it's the opposite, since this is now implemented we _are_ dropping connections so the length is now 1 (it was 0 since no connections were being dropped before). 
This is pretty confusing since the base adapter assumes a ODBC/JDBC style of connecting to a database (i.e. where you hold a connection open to a database) so a thread is equivalent to a connection. Instead with bigquery we make requests and get async jobs.

Reply:
okay got the three unit tests passing, had to incorporate some stuff from other repos and go over how we build out the `add_query` part for bigquery since it doesn't actually have that method in `connections`

My understanding of the original comment is the above was failing as we weren't targeting `key` thread identifier variable on the mock so wasn't correctly dropping in the assert